backend/src/test/java/com/bikes/backend/service/BikeServiceTest.java

```diff
@@ -3,43 +3,120 @@
+import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
-import org.springframework.stereotype.Service;
 
 import java.util.ArrayList;
 import java.util.List;
-import java.util.Map;
+import java.util.Optional;
 
-import static org.junit.jupiter.api.Assertions.*;
 import static org.mockito.Mockito.*;
+
 class BikeServiceTest {
 
-    BikeRepository bikeRepository = mock(BikeRepository.class);
-    BikeService bikeService = new BikeService(bikeRepository);
+	BikeRepository bikeRepository = mock(BikeRepository.class);
+	IdService idService = mock(IdService.class);
+    BikeService bikeService = new BikeService(bikeRepository, idService);
 
-    Bike testBike = new Bike("testBike", "testId");
-    List<Bike> expectedBikes = List.of(testBike);
-    List<Bike> expectedBikesEmpty = new ArrayList<>();
+	Bike testBike = new Bike("testId", "testBike");
+	List<Bike> expectedBikes = List.of(testBike);
+	List<Bike> expectedBikesEmpty = new ArrayList<>();
 
-    @Test
-    void getAllBikesNonEmptyRepo() {
-        //GIVEN
-        when(bikeRepository.getAllBikes()).thenReturn(expectedBikes);
-        //WHEN
-        List<Bike> actual = bikeService.getAllBikes();
-        //THEN
-        verify(bikeRepository).getAllBikes();
-        Assertions.assertEquals(expectedBikes, actual);
-    }
 
+	@Nested
+	@DisplayName("testing getAllBikes()")
+	class getAllBikesTest {
+		@Test
+		@DisplayName("...returns all bikes if the repo is not empty")
+		void getAllBikes_returnsAllBikesIfTheRepoIsNotEmpty() {
+			//GIVEN
+			when(bikeRepository.getAllBikes()).thenReturn(expectedBikes);
+			//WHEN
+			List<Bike> actual = bikeService.getAllBikes();
+			//THEN
+			verify(bikeRepository).getAllBikes();
+			Assertions.assertEquals(expectedBikes, actual);
+		}
+
+		@Test
+		@DisplayName("...returns an empty list if the repo is empty")
+		void getAllBikesEmptyRepo() {
+			//GIVEN
+			when(bikeRepository.getAllBikes()).thenReturn(expectedBikesEmpty);
+			//WHEN
+			List<Bike> actual = bikeService.getAllBikes();
+			//THEN
+			verify(bikeRepository).getAllBikes();
+			Assertions.assertEquals(expectedBikesEmpty, actual);
+		}
+
+	}
+
+	@Nested
+	@DisplayName("testing getBikeById()")
+	class getBikeByIdTest {
+
+		@Test
+		@DisplayName("...returns a bike if a bike with the given id exists")
+		void getBikeById_returnsABikeIfABikeWithTheGivenIdExists() {
+			//GIVEN
+			when(bikeRepository.getBikeById(testBike.id())).thenReturn(Optional.of(testBike));
+			//WHEN
+			Bike actual = bikeService.getBikeById(testBike.id());
+			//THEN
+			verify(bikeRepository).getBikeById(testBike.id());
+			Assertions.assertEquals(testBike, actual);
+		}
+
+		@Test
+		@DisplayName("...throws an exception if no bike with the given id exists")
+		void getBikeById_throwExceptionIfNoBikeWithTheGivenIdExists() {
+			//GIVEN
+			String invalidId = "invalidId";
+			//WHEN
+			when(bikeRepository.getBikeById(invalidId)).thenThrow(new NoSuchBikeException());
```

Comment:
Attention! Repository should not throw an Exception but instead return an empty Optional

Reply:
You're right! Thanks, deleted the line